include/Probe.php

```diff
@@ -315,7 +330,34 @@ public static function uri($uri, $network = "", $uid = 0, $cache = true) {
 	 * @return array uri data
 	 */
 	private function detect($uri, $network, $uid) {
-		if (strstr($uri, '@')) {
+		$parts = parse_url($uri);
+
+		if (isset($parts["scheme"]) AND isset($parts["host"]) AND isset($parts["path"])) {
+
+			/// @todo: Ports?
+			$host = $parts["host"];
+
+			if ($host == 'twitter.com') {
+				return array("network" => NETWORK_TWITTER);
```

Comment:
What about the `$network` parameter, what is the expected behavior when we are checking for a specific network?